cloudvolume/cloudvolume.py

```diff
@@ -261,22 +265,31 @@ def create_new_info(cls,
       chunk_size: int (x,y,z), dimensions of each downloadable 3D image chunk in voxels
       compressed_segmentation_block_size: (x,y,z) dimensions of each compressed sub-block
         (only used when encoding is 'compressed_segmentation')
+      mip_num: (int), the number of mip levels.
 
     Returns: dict representing a single mip level that's JSON encodable
     """
     info = {
       "num_channels": int(num_channels),
       "type": layer_type,
       "data_type": data_type,
-      "scales": [{
+    }
+    
+    # add mip levels
+    info['scales'] = []
+    for mip in range(mip_num):
+      resolution = (resolution[0]*2**mip, resolution[1]*2**mip, resolution[2])
```

Comment:
This is good for most use cases, but only works if your mip levels are 2x2x1 downsampled. The more authoritative logic is located in Igneous in downsample_scales. We could integrate there here maybe?

Reply:
On second thought this is fine. In the future, we should add a "factor" parameter with it set to (2,2,1) by default.

Reply:
good catch. using existing add_scale function now.